tests/failpoints/cases/test_disk_full.rs

```diff
@@ -1,281 +1,111 @@
-                            if error.has_epoch_not_match()
-                                || error.has_not_leader()
-                                || error.has_stale_command()
-                                || error
-                                    .get_message()
-                                    .contains("peer has not applied to current term")
-                            {
-                                return;
-                            }
-                            if error
-                                .get_message()
-                                .contains("disk full, all the business data write forbiden")
-                            {
-                                fg.store(false, Ordering::Release);
-                                return;
-                            }
-                            panic!("split region match unexcept error: {:?}", resp);
-                        }
-                        let admin_resp = resp.mut_admin_response();
-                        let split_resp = admin_resp.mut_splits();
-                        let regions = split_resp.get_regions();
-                        assert_eq!(regions.len(), 1);
-                    });
-                    cluster.split_region(&region, &split_key, Callback::write(check));
-                }
-
-                if cluster.pd_client.check_split(&region, &split_key)
-                    && cluster.pd_client.get_split_count() > split_count
-                {
-                    panic!("should not split when leader disk full");
-                }
-
-                if try_cnt > 250 {
-                    flag.store(false, Ordering::Release);
-                }
-                try_cnt += 1;
-                sleep(Duration::from_millis(20));
-            }
-        }
-
-        let raft_stat = cluster.raft_local_state(1, leader.get_store_id());
-        let index_3 = raft_stat.get_last_index();
-        assert!(index_1 == index_3);
-    }
-
-    {
-        // merge region not allowed.
-        let raft_stat = cluster.raft_local_state(1, leader.get_store_id());
-        let index_1 = raft_stat.get_last_index();
-
-        let left_key = String::from("5555").into_bytes();
-        let right_key = String::from("7777").into_bytes();
-        let region1 = cluster.get_region(&left_key);
-        let region2 = cluster.get_region(&right_key);
-        assert!(region1.get_id() != region2.get_id());
-        let resp = cluster.try_merge(region1.get_id(), region2.get_id());
-        assert!(
-            resp.get_header()
-                .get_error()
-                .get_message()
-                .contains("disk full, all the business data write forbiden")
-        );
-
-        let raft_stat = cluster.raft_local_state(1, leader.get_store_id());
-        let index_4 = raft_stat.get_last_index();
-        assert!(index_1 == index_4);
-    }
-
-    fail::remove(DISK_FULL_PEER);
+fn test_unallowed_leader_behaviors(cluster: &mut Cluster<ServerCluster>) {
```

Comment:
The design of these test cases has many assumptions, and the raftr state is uncertain. Based on these assumptions, the development of the use cases will be unstable, right?